### ISSUE TITLE
Driver for `application/xml`

### ISSUE THREAD

**Issue by fnattino** [intake/intake-stac#97]
Thanks a lot for this fantastic plugin, it makes it really easy to access data from STAC catalogs!

I am currently working with the public Sentinel-2 COGs collection on AWS and I have noticed an issue when trying to access the XML metadata file with Intake/Intake-STAC. The issue seems to originate from the fact that this asset is labeled as `application/xml`, but only `text/xml` seems to be present in the dictionary of drivers in Intake-STAC. Do you think it makes sense to add `'application/xml': 'textiles'`  to the dictionary of drivers? If so, I would be happy to contribute with a PR.

Thanks so much in advance! 

Here is a minimum working example:
```python
import intake

item_url = 'https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/items/S2A_31UFU_20210211_0_L2A'
item = intake.open_stac_item(item_url)
item['metadata'].read()
```

Traceback:
``` python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/opt/miniconda3/envs/test/lib/python3.7/site-packages/xarray/backends/file_manager.py in _acquire_with_cache_info(self, needs_lock)
    198             try:
--> 199                 file = self._cache[self._key]
    200             except KeyError:

/opt/miniconda3/envs/test/lib/python3.7/site-packages/xarray/backends/lru_cache.py in __getitem__(self, key)
     52         with self._lock:
---> 53             value = self._cache[key]
     54             self._cache.move_to_end(key)

KeyError: [<function open at 0x7fa073ff4f80>, ('https://roda.sentinel-hub.com/sentinel-s2-l2a/tiles/31/U/FU/2021/2/11/0/metadata.xml',), 'r', ()]

During handling of the above exception, another exception occurred:

CPLE_OpenFailedError                      Traceback (most recent call last)
rasterio/_base.pyx in rasterio._base.DatasetBase.__init__()

rasterio/_shim.pyx in rasterio._shim.open_dataset()

rasterio/_err.pyx in rasterio._err.exc_wrap_pointer()

CPLE_OpenFailedError: '/vsicurl/https://roda.sentinel-hub.com/sentinel-s2-l2a/tiles/31/U/FU/2021/2/11/0/metadata.xml' not recognized as a supported file format.

During handling of the above exception, another exception occurred:

RasterioIOError                           Traceback (most recent call last)
<ipython-input-9-760ed87ebc5b> in <module>
----> 1 item['metadata'].read()

/opt/miniconda3/envs/test/lib/python3.7/site-packages/intake_xarray/base.py in read(self)
     37     def read(self):
     38         """Return a version of the xarray with all the data in memory"""
---> 39         self._load_metadata()
     40         return self._ds.load()
     41 

/opt/miniconda3/envs/test/lib/python3.7/site-packages/intake/source/base.py in _load_metadata(self)
    234         """load metadata only if needed"""
    235         if self._schema is None:
--> 236             self._schema = self._get_schema()
    237             self.dtype = self._schema.dtype
    238             self.shape = self._schema.shape

/opt/miniconda3/envs/test/lib/python3.7/site-packages/intake_xarray/raster.py in _get_schema(self)
    100 
    101         if self._ds is None:
--> 102             self._open_dataset()
    103 
    104             ds2 = xr.Dataset({'raster': self._ds})

/opt/miniconda3/envs/test/lib/python3.7/site-packages/intake_xarray/raster.py in _open_dataset(self)
     89         else:
     90             self._ds = xr.open_rasterio(files, chunks=self.chunks,
---> 91                                         **self._kwargs)
     92 
     93     def _get_schema(self):

/opt/miniconda3/envs/test/lib/python3.7/site-packages/xarray/backends/rasterio_.py in open_rasterio(filename, parse_coordinates, chunks, cache, lock)
    274 
    275     manager = CachingFileManager(rasterio.open, filename, lock=lock, mode="r")
--> 276     riods = manager.acquire()
    277     if vrt_params is not None:
    278         riods = WarpedVRT(riods, **vrt_params)

/opt/miniconda3/envs/test/lib/python3.7/site-packages/xarray/backends/file_manager.py in acquire(self, needs_lock)
    179             An open file object, as returned by ``opener(*args, **kwargs)``.
    180         """
--> 181         file, _ = self._acquire_with_cache_info(needs_lock)
    182         return file
    183 

/opt/miniconda3/envs/test/lib/python3.7/site-packages/xarray/backends/file_manager.py in _acquire_with_cache_info(self, needs_lock)
    203                     kwargs = kwargs.copy()
    204                     kwargs["mode"] = self._mode
--> 205                 file = self._opener(*self._args, **kwargs)
    206                 if self._mode == "w":
    207                     # ensure file doesn't get overriden when opened again

/opt/miniconda3/envs/test/lib/python3.7/site-packages/rasterio/env.py in wrapper(*args, **kwds)
    433 
    434         with env_ctor(session=session):
--> 435             return f(*args, **kwds)
    436 
    437     return wrapper

/opt/miniconda3/envs/test/lib/python3.7/site-packages/rasterio/__init__.py in open(fp, mode, driver, width, height, count, crs, transform, dtype, nodata, sharing, **kwargs)
    218         # None.
    219         if mode == 'r':
--> 220             s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
    221         elif mode == "r+":
    222             s = get_writer_for_path(path, driver=driver)(

rasterio/_base.pyx in rasterio._base.DatasetBase.__init__()

RasterioIOError: '/vsicurl/https://roda.sentinel-hub.com/sentinel-s2-l2a/tiles/31/U/FU/2021/2/11/0/metadata.xml' not recognized as a supported file format.
```

The output of `item['metadata'].metadata`:
```python
{'title': 'Original XML metadata',
 'type': 'application/xml',
 'roles': ['metadata'],
 'href': 'https://roda.sentinel-hub.com/sentinel-s2-l2a/tiles/31/U/FU/2021/2/11/0/metadata.xml',
 'catalog_dir': ''}
```

"Hacking" the drivers' dictionary allows to correctly retrieve the file:
```python
import intake_stac
intake_stac.catalog.drivers['application/xml'] = 'textfiles'

import intake
item = intake.open_stac_item(item_url)
item['metadata'].read()
```

I am using Intake version 0.6.2 and Intake-STAC version 0.3.0.

**Comment by scottyhq** [comment]
Thanks for documenting this so clearly @fnattino , glad you have a workaround and a PR would be welcome!

**Comment by TomAugspurger** [comment]
Closed by #98.
